src/main/java/jenkins/plugins/purgejobhistory/PurgeJobHistory.java

```diff
@@ -0,0 +1,49 @@
+package jenkins.plugins.purgejobhistory;
+
+import hudson.Extension;
+import hudson.cli.CLICommand;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TopLevelItem;
+import java.util.ArrayList;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+/**
+ * @author Stephen Connolly
+ */
+@Extension
+public class PurgeJobHistory extends CLICommand {
+    
+    /**
+     * Follows the progress of the operation
+     */
+    @Option(name = "-r", usage = "Also reset the next build number.")
+    public boolean reset = false;
+
+    /**
+     * The source item.
+     */
+    @Argument(metaVar = "JOB", usage = "Name of the job to purge", required = true)
+    public TopLevelItem job;
+
+    
+    @Override
+    public String getShortDescription() {
+        return Messages.PurgeJobHistory_ShortDescription();
+    }
+
+    @Override
+    protected int run() throws Exception {
+        job.checkPermission(Item.DELETE);
```

Comment:
Does not make sense. Rather use

```java
run.checkPermission(Run.DELETE);
```

in the loop. (Or if that proves too expensive, at least on the first iteration, assuming [this default](https://github.com/jenkinsci/jenkins/blob/c3bc2214dbb9648294acb994e54ee84a6c043321/core/src/main/java/hudson/model/Run.java#L1455) is not overridden.)